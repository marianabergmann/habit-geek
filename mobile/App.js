import React from "react";
import { WebView } from "react-native";

export default function App() {
  return (
    <WebView source={{ uri: "https://github.com/facebook/react-native" }} />
  );
}
