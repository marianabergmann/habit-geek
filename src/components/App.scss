.mobile-size-container {
  max-width: 340px;
  padding: 15px;
  margin: auto;
}

.large-size-container {
  max-width: 600px;
  padding: 15px;
  margin: auto;
}

.mobile-size-container,
.large-size-container {
  h2 {
    text-align: center;
  }
}
