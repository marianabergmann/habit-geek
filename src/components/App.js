import React from "react";
import { Router } from "@reach/router";
import "./App.scss";
import WelcomeContainer from "./WelcomeContainer";
import DashboardContainer from "./DashboardContainer";

function App() {
  return (
    <div>
      <Router>
        <WelcomeContainer path="/" default></WelcomeContainer>
        <DashboardContainer path="/dashboard"></DashboardContainer>
      </Router>
    </div>
  );
}

export default App;
