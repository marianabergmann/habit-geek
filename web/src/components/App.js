import React from "react";
import { Router } from "@reach/router";

import { getCurrentWeek } from "@habit-geek/shared/utils/dateUtils";

import ProfileContainer from "./user/ProfileContainer";
import DashboardContainer from "./dashboard/DashboardContainer";
import Layout from "./layout/Layout";
import HistoryContainer from "./history/HistoryContainer";

import "./App.scss";

function App({ currentWeek, startNewWeek }) {
  if (currentWeek !== getCurrentWeek()) {
    startNewWeek();
  }

  return (
    <div>
      <Router>
        <Layout path="/">
          <DashboardContainer path="/" default></DashboardContainer>
          <ProfileContainer path="/profile"></ProfileContainer>
          <HistoryContainer path="/history"></HistoryContainer>
        </Layout>
      </Router>
    </div>
  );
}

export default App;
