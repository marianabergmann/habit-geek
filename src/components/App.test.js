import React from "react";
import { shallow } from "enzyme";
import App from "./App";

describe("App component", () => {
  it("Renders static elements", () => {
    const app = shallow(<App />);
    expect(app).toMatchSnapshot();
  });
});
