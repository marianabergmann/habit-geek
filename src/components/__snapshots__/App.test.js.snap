// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Renders static elements 1`] = `
<div>
  <Router>
    <Connect(Welcome)
      default={true}
      path="/"
    />
    <Connect(Dashboard)
      path="/dashboard"
    />
  </Router>
</div>
`;
